### PR TITLE
Migrate all eq filters

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description
Migrated all the tests to use `_EQ` rather than explicit equality. Fix `_EQ` with subscriptions.

### CODE REVIEW

File: packages/graphql/src/schema/resolvers/subscriptions/where/utils/parse-filter-property.ts [hunk 1]
```diff
@@ -17,17 +17,30 @@
...
 export function parseFilterProperty(key: string): { fieldName: string; operator: string | undefined } {
-    const match = whereRegEx.exec(key);
-    if (!match) {
-        throw new Error(`Failed to match key in filter: ${key}`);
+    // eslint-disable-next-line prefer-const
+    let { fieldName: stuff, operator: operatorStuff, isNot } = parseWhereField(key);
+
+    // These conversions are only temporary necessary until the the _NOT operator exists, after that we can just return the output of parseWhereField
+    if (operatorStuff === "EQ") {
+        operatorStuff = undefined;
```
Comment (angrykoala) [inline packages/graphql/src/schema/resolvers/subscriptions/where/utils/parse-filter-property.ts]: what is `operatorStuff`?

File: packages/graphql/src/schema/resolvers/subscriptions/where/utils/parse-filter-property.ts [hunk 2]
```diff
@@ -17,17 +17,30 @@
...
+            operatorStuff = `NOT_${operatorStuff}`;
+        } else {
+            operatorStuff = "NOT";
+        }
     }
-    return { fieldName, operator };
+    return { fieldName: stuff, operator: operatorStuff };
+}
+
+function isAPossibleNotOperator(
```
Comment (angrykoala) [inline packages/graphql/src/schema/resolvers/subscriptions/where/utils/parse-filter-property.ts]: I see that operators are prepended with `NOT_` if they are in the list defined in this function. What is special about these operators? Maybe we could reflect that in the name of the function?

File: packages/graphql/src/translate/queryAST/ast/filters/utils/create-point-operation.ts [hunk 3]
```diff
@@ -19,15 +19,15 @@
...
+import type { FilterOperator } from "../Filter";
 
 export function createPointOperation({
     operator,
     property,
     param,
     attribute,
 }: {
-    operator: WhereOperator | "EQ";
+    operator: FilterOperator;
```
Comment (angrykoala) [inline packages/graphql/src/translate/queryAST/ast/filters/utils/create-point-operation.ts:30]: Nice

File: packages/graphql/src/translate/queryAST/factory/OperationFactory.ts [hunk 4]
```diff
@@ -403,6 +403,7 @@ export class OperationsFactory {
 
         if (resolveTree.name === "_entities" && reference) {
             const { __typename, ...referenceWhere } = reference;
+            console.log(`reference where: ${JSON.stringify(referenceWhere, null, 2)}`);
```
Comment (angrykoala) [inline packages/graphql/src/translate/queryAST/factory/OperationFactory.ts]: :eyes:

File: packages/graphql/tests/integration/deprecations/implicit-eq-filtering.test.ts [hunk 5]
```diff
@@ -0,0 +1,193 @@
+/*
```
Comment (angrykoala) [inline packages/graphql/tests/integration/deprecations/implicit-eq-filtering.test.ts]: Is this file running on the tests? it is missing the `int.test.ts`

File: packages/graphql/tests/integration/deprecations/implicit-eq-filtering.test.ts [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
```
Comment (angrykoala) [inline packages/graphql/tests/integration/deprecations/implicit-eq-filtering.test.ts]: Is it worth adding a case for nested? just for completeness?

**Review by angrykoala — CHANGES_REQUESTED** [review]

File: packages/graphql/src/schema/resolvers/subscriptions/where/utils/parse-filter-property.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/src/schema/resolvers/subscriptions/where/utils/parse-filter-property.ts]: 👀  Ops, my bad!

**Review by angrykoala — APPROVED** [review]

**Pull request merged**